src/OSPSuite.Core/Comparison/DiffBuilder.cs

```diff
@@ -102,17 +102,28 @@ protected void CompareNullableDoubleValues<TInput>(Func<TInput, double?> funcEva
          CompareDoubleValues(x => funcEvaluation(x).GetValueOrDefault(double.NaN), propertyName, comparison, displayUnitFunc);
       }
 
-    
       protected void CompareDoubleValues<TInput>(Func<TInput, double> funcEvaluation, string propertyName, IComparison<TInput> comparison, Func<TInput, Unit> displayUnitFunc = null) where TInput : class
       {
-         try
-         {
-            CompareValues(funcEvaluation, propertyName, comparison, (x, y) => ValueComparer.AreValuesEqual(x, y, comparison.Settings.RelativeTolerance), (input, output) => numericFormatter(input, output, displayUnitFunc));
-         }
-         catch (OSPSuiteException)
+         // Determine the display unit to use for both values (based on object1)
+         var standardDisplayUnit = comparison.Object1 is IWithDimension && displayUnitFunc != null ? displayUnitFunc(comparison.Object1) : null;
+
+         string ConsistentUnitFormatter(TInput input, double value)
```

Comment:
I've not seen this pattern very often. Would the local function be otherwise useful if you made it a member of the class instead of limited to local?

I think this is ok to have it scope limited, but since you are only using it once, why not make it anonymous lambda

Reply:
I choose this because the name 'ConsistentUnitFormatter' documents the intention and ik keeps the CompareValues call clean and readable

While it's used only once, I think the clarity it adds outweighs the extra lines. What do you think?